core/src/main/java/com/yahoo/imapnio/async/request/ImapArgumentFormatter.java

```diff
@@ -226,4 +226,51 @@ String buildFlagString(@Nonnull final Flags flags) {
         sb.append(ImapClientConstants.R_PAREN); // terminate flag_list
         return sb.toString();
     }
+
+    /**
+     * Creates an IMAP entry-flag-name from the given Flags with only one flag object. The following is the ABNF from RFC7162
+     * <pre>
+     * entry-flag-name     = DQUOTE "/flags/" attr-flag DQUOTE
+     *                        ;; Each system or user-defined flag <flag>
+     *                        ;; is mapped to "/flags/<flag>".
+     *                        ;;
+     *                        ;; <entry-flag-name> follows the escape rules
+     *                        ;; used by "quoted" string as described in
+     *                        ;; Section 4.3 of [RFC3501]; e.g., for the
+     *                        ;; flag \Seen, the corresponding <entry-name>
+     *                        ;; is "/flags/\\seen", and for the flag
+     *                        ;; $MDNSent, the corresponding <entry-name>
+     *                        ;; is "/flags/$mdnsent".
+     * </pre>
+     *
+     * @param flags the flags with one flag
+     * @return the entry flag name string
+     */
+     String buildEntryFlagName(@Nonnull final Flags flags)  {
+        final Flags.Flag[] sf = flags.getSystemFlags(); // get the system flags
+        final String[] uf = flags.getUserFlags(); // get the user flag strings
+        final StringBuilder s = new StringBuilder("\"/flags/"); // start of entry flag name
+
+        if (sf.length == 1) {
+            s.append(ImapClientConstants.BACKSLASH);
+            if (sf[0] == Flags.Flag.ANSWERED) {
+                s.append(ANSWERED);
+            } else if (sf[0] == Flags.Flag.DELETED) {
+                s.append(DELETED);
+            } else if (sf[0] == Flags.Flag.DRAFT) {
+                s.append(DRAFT);
+            } else if (sf[0] == Flags.Flag.FLAGGED) {
+                s.append(FLAGGED);
+            } else if (sf[0] == Flags.Flag.RECENT) {
+                s.append(RECENT);
+            } else if (sf[0] == Flags.Flag.SEEN) {
+                s.append(SEEN);
+            }
+        } else { // user flag == 1
+            s.append(uf[0]);
+        }
```

Comment:
Based on getUserFlags(), it will try to find it in the HashMap. If it doesn't create the flag, it will add the flag into user flag. Since there is only one flag and the flag can only be user flag or system flag. As long as it is not system flag, it will count as user flag. You can check buildFlagString inside the same file to confirm the implementation flow.